### PR TITLE
8282215: Handle failures when initializing SystemLookup

### PR DESCRIPTION
**Pull request by mcimadamore**
This patch handles failures originating from initialization of the SystemLookup class; instead of propagating these failures, we now swallow them, so that initialization of a `CLinker` can proceed. This adds some robustness to client code, and doubles down on the optional return type of the CLinker's lookup method.

The test checks that, when running with `os.name` set to Windows, we always manage to perform a lookup (even if we are executing the test on Linux/MacOS). The test trivially passes on Windows.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282215](https://bugs.openjdk.java.net/browse/JDK-8282215): Handle failures when initializing SystemLookup

### Reviewers
 * [Jorn Vernee](https://openjdk.java.net/census#jvernee) (@JornVernee - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/649/head:pull/649` \
`$ git checkout pull/649`

Update a local copy of the PR: \
`$ git checkout pull/649` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/649/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 649`

View PR using the GUI difftool: \
`$ git pr show -t 649`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/649.diff">https://git.openjdk.java.net/panama-foreign/pull/649.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/foreign/SequenceLayout.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ public MemoryLayout elementLayout() {
     }
 
     /**
-     * {@return the element count of this sequence layout (if any)}
+     * {@return the element count of this sequence layout}
```
Comment (mcimadamore) [inline src/java.base/share/classes/java/lang/foreign/SequenceLayout.java:94]: This is an unrelated fix, addressing a javadoc issue introduced by JDK-8282026

**Review by JornVernee — APPROVED** [review]
